// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: user_center/oauth2/v2/oauth2.proto

package v2

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on AccessTokenRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *AccessTokenRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on AccessTokenRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// AccessTokenRequestMultiError, or nil if none found.
func (m *AccessTokenRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *AccessTokenRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Code

	// no validation rules for Secret

	// no validation rules for NodeId

	if len(errors) > 0 {
		return AccessTokenRequestMultiError(errors)
	}

	return nil
}

// AccessTokenRequestMultiError is an error wrapping multiple validation errors
// returned by AccessTokenRequest.ValidateAll() if the designated constraints
// aren't met.
type AccessTokenRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m AccessTokenRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m AccessTokenRequestMultiError) AllErrors() []error { return m }

// AccessTokenRequestValidationError is the validation error returned by
// AccessTokenRequest.Validate if the designated constraints aren't met.
type AccessTokenRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e AccessTokenRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e AccessTokenRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e AccessTokenRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e AccessTokenRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e AccessTokenRequestValidationError) ErrorName() string {
	return "AccessTokenRequestValidationError"
}

// Error satisfies the builtin error interface
func (e AccessTokenRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sAccessTokenRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = AccessTokenRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = AccessTokenRequestValidationError{}

// Validate checks the field values on AccessTokenReplay with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *AccessTokenReplay) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on AccessTokenReplay with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// AccessTokenReplayMultiError, or nil if none found.
func (m *AccessTokenReplay) ValidateAll() error {
	return m.validate(true)
}

func (m *AccessTokenReplay) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetAuthToken()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, AccessTokenReplayValidationError{
					field:  "AuthToken",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, AccessTokenReplayValidationError{
					field:  "AuthToken",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetAuthToken()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return AccessTokenReplayValidationError{
				field:  "AuthToken",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return AccessTokenReplayMultiError(errors)
	}

	return nil
}

// AccessTokenReplayMultiError is an error wrapping multiple validation errors
// returned by AccessTokenReplay.ValidateAll() if the designated constraints
// aren't met.
type AccessTokenReplayMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m AccessTokenReplayMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m AccessTokenReplayMultiError) AllErrors() []error { return m }

// AccessTokenReplayValidationError is the validation error returned by
// AccessTokenReplay.Validate if the designated constraints aren't met.
type AccessTokenReplayValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e AccessTokenReplayValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e AccessTokenReplayValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e AccessTokenReplayValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e AccessTokenReplayValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e AccessTokenReplayValidationError) ErrorName() string {
	return "AccessTokenReplayValidationError"
}

// Error satisfies the builtin error interface
func (e AccessTokenReplayValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sAccessTokenReplay.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = AccessTokenReplayValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = AccessTokenReplayValidationError{}

// Validate checks the field values on QrConnectRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *QrConnectRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on QrConnectRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// QrConnectRequestMultiError, or nil if none found.
func (m *QrConnectRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *QrConnectRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for NodeId

	// no validation rules for RedirectUri

	// no validation rules for ResponseType

	// no validation rules for Scope

	// no validation rules for State

	if len(errors) > 0 {
		return QrConnectRequestMultiError(errors)
	}

	return nil
}

// QrConnectRequestMultiError is an error wrapping multiple validation errors
// returned by QrConnectRequest.ValidateAll() if the designated constraints
// aren't met.
type QrConnectRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m QrConnectRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m QrConnectRequestMultiError) AllErrors() []error { return m }

// QrConnectRequestValidationError is the validation error returned by
// QrConnectRequest.Validate if the designated constraints aren't met.
type QrConnectRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e QrConnectRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e QrConnectRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e QrConnectRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e QrConnectRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e QrConnectRequestValidationError) ErrorName() string { return "QrConnectRequestValidationError" }

// Error satisfies the builtin error interface
func (e QrConnectRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sQrConnectRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = QrConnectRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = QrConnectRequestValidationError{}

// Validate checks the field values on QrConnectReply with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *QrConnectReply) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on QrConnectReply with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in QrConnectReplyMultiError,
// or nil if none found.
func (m *QrConnectReply) ValidateAll() error {
	return m.validate(true)
}

func (m *QrConnectReply) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return QrConnectReplyMultiError(errors)
	}

	return nil
}

// QrConnectReplyMultiError is an error wrapping multiple validation errors
// returned by QrConnectReply.ValidateAll() if the designated constraints
// aren't met.
type QrConnectReplyMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m QrConnectReplyMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m QrConnectReplyMultiError) AllErrors() []error { return m }

// QrConnectReplyValidationError is the validation error returned by
// QrConnectReply.Validate if the designated constraints aren't met.
type QrConnectReplyValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e QrConnectReplyValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e QrConnectReplyValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e QrConnectReplyValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e QrConnectReplyValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e QrConnectReplyValidationError) ErrorName() string { return "QrConnectReplyValidationError" }

// Error satisfies the builtin error interface
func (e QrConnectReplyValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sQrConnectReply.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = QrConnectReplyValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = QrConnectReplyValidationError{}
